src/libsyntax/parse/parser.rs

```diff
@@ -6172,6 +6174,45 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub fn parse_expansion(&mut self, kind: ExpansionKind) -> PResult<'a, Expansion> {
```

Comment:
I'm kind of wary of this because it introduces new types to the parser module and it doesn't seem to need to be a method - it doesn't use any of Parser's private API afaict, so seems like it would be fine in expand.rs